tests/mcmc/test_proposal.py

```diff
@@ -65,3 +74,75 @@ def _check_mean_and_std(expected_mean, expected_std, samples):
             rtol=1e-2,
             atol=1e-1,
         )
+
+
+class TestTransitionAwareProposalGenerator(unittest.TestCase):
```

Comment:
Use `chex.TestCase` as base class for tests. This way we are able to tests important JAX usage like jitting, pmapping and using device memory for GPU usage. 

Same for all new tests here and in `test_random_walk.py`